meteor/server/api/rest/v1/playlists.ts

```diff
@@ -210,6 +213,57 @@ class PlaylistsServerAPI implements PlaylistsRestAPI {
 			)
 		}
 	}
+	async executeBucketAdLib(
+		connection: Meteor.Connection,
+		event: string,
+		rundownPlaylistId: RundownPlaylistId,
+		bucketId: BucketId,
+		externalId: string,
+		triggerMode?: string | null
+	): Promise<ClientAPI.ClientResponse<object>> {
+		const bucketPromise = Buckets.findOneAsync({}, { projection: { _id: 1, name: 1, studioId: 1 } })
+		const bucketAdlibPromise = BucketAdLibs.findOneAsync({ bucketId, externalId }, { projection: { _id: 1 } })
```

Comment:
I think it would be better to `await Promise.all(` these promises.

If a promise rejects but it was not awaited or had `.catch(` used on it, then it becomes an unhandledPromiseRejection and can result in an application crash

Reply:
I also think I used the wrong selector for the buckets here...

Reply:
Updated